### PR TITLE
[feat] support generic via deriving

### PR DESCRIPTION
**Pull request by MangoIV**
- support generic via deriving ala 
  `deriving (FromForm, ToForm) via Generically MyType`

### CODE REVIEW

File: http-api-data.cabal [hunk 1]
```diff
@@ -47,6 +47,8 @@ library
                    , containers            >= 0.5.10.2 && < 0.7
                    , text                  >= 1.2.3.0  && < 1.3 || >=2.0 && <2.1
                    , transformers          >= 0.5.2.0  && < 0.7
+    if impl(ghc >= 9.4.1)
+    build-depends:   generically
```
Comment (phadej) [inline http-api-data.cabal:51]: This looks wrong

**Review by phadej — CHANGES_REQUESTED** [review]

File: http-api-data.cabal (hunk 1 above)
Comment (MangoIV) [reply http-api-data.cabal:51]: Pups yes. >= is the wrong way. Why did everything build though? :D

Comment (phadej) [reply http-api-data.cabal:51]: CI is not run until I approve this PR, and I didn't bother. Please try locally with GHC-9.2 or older.

Comment (MangoIV) [reply http-api-data.cabal:51]: It runs on my fork… But yeah the reason is probably that the check is ignored

Comment (MangoIV) [reply http-api-data.cabal]: Seems like it’s fixed now :)

**Comment by MangoIV** [comment]
Hi, I think I have addressed your issues with my PR, @phadej, would you mind proceeding?

**Comment by MangoIV** [comment]
@phadej just pinging again in case you overlooked this.

**Pull request closed without merging**